src/glib/base/dt.cpp

```diff
@@ -2207,13 +2207,29 @@ TStr TStr::GetStr(const TStrV& StrV, const TStr& DelimiterStr){
   return ResStr;
 }
 
-TStr TStr::Fmt(const char *FmtStr, ...){
-  char Bf[10*1024];
-  va_list valist;
-  va_start(valist, FmtStr);
-  const int RetVal=vsnprintf(Bf, 10*1024-2, FmtStr, valist);
-  va_end(valist);
-  return RetVal!=-1 ? TStr(Bf) : TStr();
+TStr TStr::Fmt(const char *FmtStr, ...) {
+    char Bf[10 * 1024];
+    va_list valist;
+    va_start(valist, FmtStr);
+    const int RequiredLen = vsnprintf(Bf, 10 * 1024 - 2, FmtStr, valist);
+    va_end(valist);
+    const int Len = strlen(Bf);
```

Comment:
Here we could avoid strlen by comparing against 10*1024-1. We only care if we could fit the string inside the buffer, not how long it actually is.